tests/tests.vala

```diff
@@ -0,0 +1,238 @@
+/*
+* Copyright (c) 2019 Alecaddd (http://alecaddd.com)
+*
+* This file is part of Akira.
+*
+* Akira is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+
+* Akira is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+
+* You should have received a copy of the GNU General Public License
+* along with Akira.  If not, see <https://www.gnu.org/licenses/>.
+*
+* Authored by: Alessandro "Alecaddd" Castellani <castellani.ale@gmail.com>
+*/
+
+public delegate void Akira.TestCaseFunc ();
+public GLib.SettingsSchemaSource schema_source;
+
+private class Akira.TestSuiteAdaptor {
+    public string name;
+
+    private Akira.TestCaseFunc func;
+    private Akira.TestSuite test_suite;
+
+    public TestSuiteAdaptor (string name, owned Akira.TestCaseFunc test_case_func,
+                             Akira.TestSuite test_suite) {
+        this.name = name;
+        this.func = (owned) test_case_func;
+        this.test_suite = test_suite;
+    }
+
+    public void setup (void* fixture) {
+        this.test_suite.setup ();
+    }
+
+    public void run (void* fixture) {
+        this.func ();
+    }
+
+    public void teardown (void* fixture) {
+        this.test_suite.teardown ();
+    }
+
+    public GLib.TestCase get_g_test_case () {
+        return new GLib.TestCase (this.name,
+                                  this.setup,
+                                  this.run,
+                                  this.teardown);
+    }
+}
+
+public abstract class Akira.TestSuite : GLib.Object {
```

Comment:
This do not need to be abstract. It complains on new vala versions

Reply:
Updated